mesonbuild/interpreter.py

```diff
@@ -587,8 +587,15 @@ def extract_objects_method(self, args, kwargs):
         gobjs = self.held_object.extract_objects(args)
         return GeneratedObjectsHolder(gobjs)
 
+    @permittedMethodKwargs({'recursive'})
     def extract_all_objects_method(self, args, kwargs):
-        gobjs = self.held_object.extract_all_objects()
+        recursive = kwargs.get('recursive', False)
+        gobjs = self.held_object.extract_all_objects(recursive)
+        if gobjs.objlist and not 'recursive' in kwargs:
```

Comment:
__[Flake8]__

[E713] test for membership should be 'not in'


<!-- Comment by [SideCI](https://sideci.com) -->